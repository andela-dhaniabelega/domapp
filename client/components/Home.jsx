import React from 'react';
import SignUpForm from './SignUpForm';

class Home extends React.Component {
	render(){
		return(
			<div>
				<h1>Home page</h1>
			</div>
		);
	}
}
export default Home;
