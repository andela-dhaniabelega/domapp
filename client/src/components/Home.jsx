import React from 'react';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import SignUpForm from './SignUpForm';
import { userSignUpRequest } from '../actions/signUpActions';
/**
 * @description Markup for Home Page
 * @return {void}
 */
const Home = () => ({
	render() {
		return (
			<div className="main">
				<div className="content">
					<h1>Domapp provides an easy solution for managing your documents. </h1>
					<p className="sub-heading">
						If you&apos;re looking for an intuitive, simple, effective
						way to manage your documents then Doomap is your choice.
					</p>
					<div className="mid-column">
						<div className="mid-column-2" />
						<div className="mid-column-1">
							<SignUpForm userSignUpRequest={this.props.userSignUpRequest} />
						</div>
					</div>
					<div className="features">
						<p className="intro">
							With Domapp document management is hassle-free
						</p>
						<div className="row f-center center-icons">
							<div className="col s4">
								<div className="round-1">
									<i className="large material-icons center-icon">view_module</i>
								</div>
								<div className="feature-text">
									<p className="main-text">Manage Docs</p>
									<p className="sub-text">Create, Read, Update and Delete Documents</p>
								</div>
							</div>
							<div className="col s4">
								<div className="round-2">
									<i className="large material-icons center-icon">lock</i>
								</div>
								<div className="feature-text">
									<p className="main-text">Granular Access</p>
									<p className="sub-text">Assign Roles to Documents and Users</p>
								</div>
							</div>
							<div className="col s4">
								<div className="round-3">
									<i className="large material-icons center-icon">assignment_ind</i>
								</div>
								<div className="feature-text">
									<p className="main-text">Manage Users</p>
									<p className="sub-text">Manage User roles and user accounts</p>
								</div>
							</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
});
Home.propTypes = {
	userSignUpRequest: propTypes.func.isRequired
};
export default connect(null, { userSignUpRequest })(Home);
