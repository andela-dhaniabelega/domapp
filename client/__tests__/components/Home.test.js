import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Home from '../../src/components/Home';

const mockStore = configureStore();
const store = mockStore({
	auth: { isAuthenticated: true, user: {} },
	users: { isAuthenticated: false, users: {}, allUsers: {}, userSearch: {}, pagination: [] },
	userDocuments: { documents: [{}], searchResults: [], document: {}, pagination: {}, errors: {} }
});
describe('Home', () => {
	it('renders a snapshot', () => {
		const wrapper = shallow(
			<Provider store={store}>
				<Home />
			</Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
