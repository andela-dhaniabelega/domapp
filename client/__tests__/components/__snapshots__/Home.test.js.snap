// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Connect(Home) />,
  "nodes": Array [
    <Connect(Home) />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Provider
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
>
        <Connect(Home) />
</Provider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Provider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <Connect(Home) />,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Connect(Home) />,
        "_debugID": 2,
        "_renderedOutput": <Connect(Home) />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Provider
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
>
    <Connect(Home) />
</Provider>,
}
`;
